### PR TITLE
feat: improve tool use response format

### PR DESCRIPTION
**Pull request by jrmi**
This is an attempt to match the expected output for each API regarding tool use response.

It modify the message dicts to match the expected output.

Also fix tool call not working with stream == False

### CODE REVIEW

File: gptme/cli.py [hunk 1]
```diff
@@ -187,8 +187,10 @@ def main(
 
     config = get_config()
 
-    tool_format = tool_format or config.get_env("TOOL_FORMAT") or "markdown"
-    set_tool_format(tool_format)
+    selected_tool_format: ToolFormat = (
+        tool_format or config.get_env("TOOL_FORMAT") or "markdown"  # type: ignore
```
Comment (jrmi) [inline gptme/cli.py:191]: Unrelated but the env variable was ignored because of the default value.

File: gptme/llm/llm_anthropic.py [hunk 2]
```diff
@@ -66,9 +67,17 @@ def chat(messages: list[Message], model: str, tools: list[ToolSpec] | None) -> s
     )
     content = response.content
     logger.debug(response.usage)
-    assert content
-    assert len(content) == 1
-    return content[0].text  # type: ignore
+
+    parsed_block = []
```
Comment (jrmi) [inline gptme/llm/llm_anthropic.py:69]: Tool call was broken with non stream API.

File: gptme/llm/llm_anthropic.py [hunk 3]
```diff
@@ -99,7 +108,7 @@ def stream(
                     block = chunk.content_block
                     if isinstance(block, anthropic.types.ToolUseBlock):
                         tool_use = block
-                        yield f"\n@{tool_use.name}: "
+                        yield f"\n@{tool_use.name}({tool_use.id}): "
```
Comment (jrmi) [inline gptme/llm/llm_anthropic.py:108]: We add the tool uid here to save it in the message.

File: gptme/llm/llm_anthropic.py [hunk 4]
```diff
@@ -224,7 +303,7 @@ def _transform_system_messages(
 
     # for any subsequent system messages, transform them into a <system> message
     for i, message in enumerate(messages):
-        if message.role == "system":
+        if message.role == "system" and message.call_id is None:
```
Comment (jrmi) [inline gptme/llm/llm_anthropic.py:305]: We don't want to change messages with call_id. They will be handled by the `_handle_tool` function.

**Comment by jrmi** [comment]
Hey @ErikBjare this MR is ready to review:
- it fixes the tool response to match the expected format by adding the `call_id` to the message
- it fixes the tool call with non-stream API
- it improves openai msg2dict process by adding the `_prepare_messages_for_api`
- it adds tests for message preparation process
- it fixes small bugs

Let me know what you think.

FYI I tested it with openai and anthropic ;-)

File: gptme/llm/llm_anthropic.py (hunk 4 above)
Comment (ErikBjare) [reply gptme/llm/llm_anthropic.py:305]: Add this as a code comment, as it's good to know when reading

Comment (jrmi) [reply gptme/llm/llm_anthropic.py:305]: Done.

**Review by ErikBjare — APPROVED** [review]
Super impressive, works amazingly! :muscle: 

Thanks again :pray: 

:1st_place_medal: :crown:

**Pull request merged**